### DISCUSSION TITLE
How to custom output file naming in nuxt3 ?

### DISCUSSION THREAD

**Discussion by Michael0520** [nuxt/nuxt#24735] [Questions (Nuxt 2)]
Hello everyone,

In Nuxt 2, I used the following configuration to control the naming of output files:

```ts
filenames: {
  app: () => '[name].js',
  chunk: () => '[name].js',
  css: () => '[name].css',
  img: () => '[path][name].[ext]',
  font: () => '[path][name].[ext]',
  video: () => '[path][name].[ext]'
}
```

(Source: [Nuxt 2 Documentation](https://v2.nuxt.com/docs/configuration-glossary/configuration-build/#filenames))

However, in Nuxt 3, I would like to achieve a similar configuration to control the naming of output files without including hash values. Could you please guide me on what settings to use in Nuxt 3 to achieve this?

Thank you all for your assistance!

**Comment by OnlyWick** [comment]
Give you some relevant information.
see this: 
- https://nuxt.com/docs/api/nuxt-config#transpile
- https://nuxt.com/docs/api/nuxt-config#vite

```typescript
/**
   * If you want to transpile specific dependencies with Babel, you can add them here. Each item in transpile can be a package name, a function, a string or regex object matching the dependency's file name.
   * 
   * You can also use a function to conditionally transpile. The function will receive an object ({ isDev, isServer, isClient, isModern, isLegacy }).
   * 
   * 
   * @example
   * ```js
   *      transpile: [({ isLegacy }) => isLegacy && 'ky']
   * ```
  */
transpile: Array<string | RegExp | ((ctx: { isClient?: boolean; isServer?: boolean; isDev: boolean }) => string | RegExp | false)>,
```

```typescript
build: {
  transpile: [({ isDev, isClient, isServer }) => {
    // ...
    return ...
  }]
},
```

and

```typescript
vite: {
  build: {
    rollupOptions: {
      output: {
        // ...
      }
    }
  }
},
```

**Comment by Michael0520** [comment]
@OnlyWick 

Hi, thank you for your response. I have made some progress and would like your assistance in reviewing the issues. I hope you can provide me with some guidance or suggestions.

lastest version :https://github.com/nuxt/nuxt/discussions/25092
